bootstrap/golang/prep-new-repo.sh

```diff
@@ -0,0 +1,109 @@
+  echo "Error: Output dirpath to copy to '${output_dirpath}' is nonexistent" >&2
+  exit 1
+fi
+
+new_image_name="${3:-}"
+if [ -z "${new_image_name}" ]; then
+  echo "Error: Empty Docker image name" >&2
+  exit 1
+fi
+
+# =============================================================================
+#                               Copying Files
+# =============================================================================
+cp "${input_dirpath}/.dockerignore" "${output_dirpath}/"
+cp "${input_dirpath}/${GO_MOD_FILENAME}" "${output_dirpath}/"
+cp "${input_dirpath}/go.sum" "${output_dirpath}/"
+cp -r "${input_dirpath}/${LAMBDA_IMPL_DIRNAME}" "${output_dirpath}/"
+cp -r "${input_dirpath}/${SCRIPTS_DIRNAME}" "${output_dirpath}/"
+
+# =============================================================================
+#                         Post-Copy Modifications
+# =============================================================================
+# Allow setting the module name programmatically, for testing in CI
+new_module_name="${GO_NEW_MODULE_NAME:-}"
+echo "Go uses Github as a its dependency management store. You'll now need to:"
+echo "  1) Create a new Github repo to contain your Kurtosis Lambda, if you don't have one already"
+echo "  2) Enter the URL of the repo on Github WITHOUT the leading 'https://' below (e.g. 'github.com/my-org/my-repo')"
+echo "NOTE: This value is technically the Go module name. If you're unfamiliar with what this is, you can read more here: https://golang.org/ref/mod"
+echo ""
+while [ -z "${new_module_name}" ]; do
+  read -p "Module name: " new_module_name
+done
+
+# Validation, to save us in case someone changes stuff in the future
+go_mod_filepath="${output_dirpath}/${GO_MOD_FILENAME}"
+if [ "$(grep -c "^${GO_MOD_MODULE_KEYWORD}" "${go_mod_filepath}")" -ne 1 ]; then
+  echo "Validation failed: Could not find exactly one line in ${GO_MOD_FILENAME} with keyword '${GO_MOD_MODULE_KEYWORD}' for use when replacing with the user's module name" >&2
+  exit 1
+fi
+
+# Replace module names in code (we need the "-i '' " argument because Mac sed requires it)
+existing_module_name="$(grep "module" "${go_mod_filepath}" | awk '{print $2}')"
+if ! sed -i"${SED_INPLACE_FILE_SUFFIX}" "s,${existing_module_name},${new_module_name},g" ${go_mod_filepath}; then
+  echo "Error: Could not replace Go module name in mod file '${go_mod_filepath}'" >&2
+  exit 1
+fi
+# We search for old_module_name/kurtosis-lambda because we don't want the old_module_name/lib entries to get renamed
+if ! sed -i"${SED_INPLACE_FILE_SUFFIX}" "s,${existing_module_name}/${LAMBDA_IMPL_DIRNAME},${new_module_name}/${LAMBDA_IMPL_DIRNAME},g" $(find "${output_dirpath}" -type f); then
+  echo "Error: Could not replace Go module name in code files" >&2
+  exit 1
+fi
+
+# Replacing Docker image name in build script
+# Validation, to save us in case someone changes stuff in the future
+image_name_replacement_pattern="^${BUILD_SCRIPT_IMAGE_NAME_VAR_NAME}=\".*\"$"
```

Comment:
This logic still needs to be moved into `bootstrap.sh`, rather than `golang/prep-new-repo.sh`, because it's not language-specific